lib/capybara/selector.rb

```diff
@@ -380,15 +381,21 @@
   end
 
   filter(:selected) do |node, selected|
-    actual = node.all(:xpath, './/option', visible: false).select { |option| option.selected? }.map { |option| option.text(:all) }
-    [selected].flatten.sort == actual.sort
+    actual = node.all(:xpath, './/option', visible: false).select(&:selected?).map { |option| option.text(:all) }
+    Array(selected).sort == actual.sort
```

Comment:
That's true, but as you said, it doesn't make a difference in this case since, based on do documentation and specs I read, nesting options is not something that's deliberately supported.